root/static/scripts/edit/URLCleanup.js

```diff
@@ -429,6 +434,53 @@ const CLEANUPS = {
       return /^https:\/\/www\.amazon\.(com|ca|co\.uk|fr|at|de|it|co\.jp|jp|cn|es|in|com\.br|com\.mx|com\.au)\//.test(url);
     },
   },
+  'amazonmusic': {
+    match: [new RegExp('^(https?://)?music\\.amazon\\.(com|ca|co\\.uk|fr|at|de|it|co\\.jp|jp|cn|es|in|com\\.br|com\\.mx|com\\.au)/(albums|artists)', 'i')],
+    type: LINK_TYPES.streaming,
+    clean: function (url) {
+      /*
+       * determine tld, asin from url, and build standard format [1],
+       * if both were found.
+       *
+       * [1] "https://www.amazon.<tld>/(albums|artists)/<ASIN>"
```

Comment:
Probably yes for the tld bit. Not sure for the rest, since I'm also taking the type this time unless on the ASIN version. Still worth it? :)

Reply:
Sure, please deduplicate what can be. A constant for Amazon public suffixes/effective TLDs would probably help with.

Reply:
Extracted a findAmazonTld function.

The tld constant is not so trivial because we need to escape dots with \ in some places and \\ in others, unless there's something I'm missing that makes that trivial too :)